narwhals/_sql/expr.py

```diff
@@ -385,7 +385,29 @@ def round(self, decimals: int) -> Self:
         return self._with_elementwise(
             lambda expr: self._function("round", expr, self._lit(decimals))
         )
-
+    
+    def _sqrt(self) -> Self:
+        return self._with_elementwise(lambda expr: self._function("isnull", expr))
+    
+    # Self: because we're giving it a column and return a column?
+    def sqrt(self) -> Self: 
+        # Native.. type because we're passing it a native function?, my instinct was to pass 'self' too, but that didn't work either
+        def _sqrt(expr: NativeExprT) -> NativeExprT: 
+            # why _lit below? is it because they're fixed. I can see that 'self' is wrong, but didn't find how to fix
+            return self._when(self < self._lit(0), self._lit(float("nan")), self._with_elementwise(lambda expr: self._function("sqrt", expr)))
+
+        return self   
```

Comment:
`Self` means that it returns an object of the same class as the input. So for example if you have `expr` and `expr` is of type `SQLExpr`, then `expr.sqrt()` is also of type `SQLExpr`

> I can see that 'self' is wrong, but didn't find how to fix

I'd suggest taking a closer look at the code in `DuckDBExpr` which you commented out